pcdsdevices/areadetector/detectors.py

```diff
@@ -20,15 +20,59 @@
 
 
 __all__ = ['PCDSAreaDetectorBase',
+           'PCDSAreaDetectorEmbedded',
            'PCDSAreaDetector']
 
 
 class PCDSAreaDetectorBase(DetectorBase):
     """Standard area detector with no plugins."""
     cam = ADComponent(cam.CamBase, '')
 
+    def get_plugin_graph_edges(self, *, use_names=True, include_cam=False):
+        """
+        Get a list of (source, destination) ports for all plugin chains.
+
+        Parameters
+        ----------
+        use_names : bool, optional
+            By default, the ophyd names for each plugin are used. Set this to
+            False to instead get the AreaDetector port names.
+        include_cam : bool, optional
+            Include plugins with 'CAM' as the source.  As it is easy to assume
+            that a camera without an explicit source is CAM, by default this
+            method does not include it in the list.
+        """
+
+        cam_port = self.cam.port_name.get()
+        graph, port_map = self.get_asyn_digraph()
+        port_edges = [(src, dest) for src, dest in graph.edges
+                      if src != cam_port or include_cam]
+        if use_names:
+            port_edges = [(port_map[src].name, port_map[dest].name)
+                          for src, dest in port_edges]
+        return port_edges
+
+
+class PCDSAreaDetectorEmbedded(PCDSAreaDetectorBase):
+    """
+    Minimal area detector including only the most-used PCDS plugins.
+
+    The plugins included are:
+        IMAGE2: reduced rate image, used for camera viewer.
+        Stats2: reduced rate stats.
+    """
```

Comment:
To clarify: pydocstyle calls for a newline after class docstrings but _not_ function docstrings.

Reply:
My/our rules call for a blank line after any multiline docstring :/

Reply:
Respective warnings are:
```
D204: 1 blank line required after class docstring (found 0)
D202: No blank lines allowed after function docstring (found 1)
```

Options would be to: tweak the style guide to reflect that or when we use pydocstyle ignore the above?

Reply:
Yeah, either way. Personally, I think the way described in the style guide looks cleaner, but I'm ok with it changing. I never implemented pydocstyle because I never got around to figuring out which warnings to use and which to ignore.

Reply:
My brain actually tunes out the bottom `"""` and treats it as a blank line... so when we add the extra line my brain sees two blank lines.

I don't mind any which way, I'll adjust this one as suggested.